buildbot/buildbot/worker.py

```diff
@@ -494,9 +492,9 @@ def stage_fpga_execute(db, config):
                 exe_cmd = ['sudo', 'sh', '-c',
                            'source /opt/xilinx/xrt/setup.sh ; ./host']
             else:
-                exe_cmd = ['sh', '-c', 'cur=`pwd`; cd $AWS_FPGA_REPO_DIR ;\
-                source ./sdaccel_setup.sh > /dev/null; \
-                cd $cur; XCL_EMULATION_MODE={} ./host'.format(task['mode'])]
+                exe_cmd = ['sh', '-c', 'source $AWS_FPGA_REPO_DIR/sdaccel_setup.sh > /dev/null;\
+                XCL_EMULATION_MODE={} ./host'.format(task['mode'])]
```

Comment:
I think the reason we need the `cur=... ; ...; cd $cur` dance is that the setup script assumes it will be `source`'d from its containing directory. But perhaps I'm wrong—does this work without the `cd` dance?

Reply:
Unrelatedly, it occurs to me that we could probably pass `XCL_EMULATION_MODE` via an `env` parameter to `task.run` as opposed to interpolating it into the shell line.

Reply:
I just did the experiment on f1 instance and it seems like the setup script does not need to be in the directory where it is sourced from! 😲

Reply:
Excellent news!

Reply:
Amazing! We can merge it whenever @Vivi verifies this doesn’t break everything (doesn’t have to be right now!)